FirebaseDynamicLinks/Sources/FIRDynamicLinkNetworking.m

```diff
@@ -101,6 +102,50 @@ - (instancetype)initWithAPIKey:(NSString *)APIKey
   return self;
 }
 
++ (NSError *)genericErrorForUrl:(NSURL *)url {
+  return [NSError errorWithDomain:kGenericErrorDomain
+                             code:0
+                         userInfo:@{
+                           @"message" : [NSString
+                               stringWithFormat:@"Failed to resolve link: %@", url.absoluteString]
+                         }];
+}
+
++ (nullable NSError *)extractErrorForShortLink:(NSURL *)url
+                                          data:(NSData *)data
+                                      response:(NSURLResponse *)response
+                                         error:(nullable NSError *)error {
+  if (error) {
+    return error;
+  }
+
+  NSInteger statusCode = [(NSHTTPURLResponse *)response statusCode];
+  NSError *customError = nil;
+
+  if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+    customError =
+        [NSError errorWithDomain:kGenericErrorDomain
+                            code:0
+                        userInfo:@{@"message" : @"Response should be of type NSHTTPURLResponse."}];
+  } else if (statusCode < 200 || statusCode >= 300) {
+    if (data) {
```

Comment:
Can you move this data check to parent `else if` so that you don't need to repeat 
     
    customError = [FIRDynamicLinkNetworking genericErrorForUrl:url];

Reply:
Addressed in https://github.com/firebase/firebase-ios-sdk/pull/5853/commits/11e614d0292f1369947f2e65e4cf4d4873a5ff65.